boards/common/openocd_segger_rtt.cfg

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2024 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+proc version_compare {v1 v2} {
+	foreach n1 [split $v1 .] n2 [split $v2 .] {
+		if {$n1 < $n2} {return -1}
+		if {$n1 > $n2} {return 1}
+	}
+	return 0
+}
+
+set current_version [version]
+regexp {([0-9]+\.[0-9]+\.[0-9]+)} $current_version -> current_version
+
+if {[version_compare $current_version "0.12.0"] >= 0} {
+	# OpenOCD version is 0.12.0 or newer - initialize Segger RTT server
+	init
+	rtt setup 0x20000000 10000 "SEGGER RTT"
+	rtt start
+	rtt server start 9090 0
```

Comment:
I like the effort that you've gone to solve the issue you encountered. However, there are a couple of reasons why we cannot merge this:

1. the setup line will only work for a small subset of SoCs. Particularly those that share an SRAM region in the same location as some stm32 parts
2. the solution in [openocd.py](https://github.com/zephyrproject-rtos/zephyr/blob/e873d503a908ab5a313032bf59c7edfe9096a002/scripts/west_commands/runners/openocd.py#L387) searches a 16-byte range for the "SEGGER RTT" identifier, but this one searches a 64kiB region of memory that may or may not exist

Reply:
Regarding other SoC I have of course less experience. Thank you for the explanation. 
I will take time and try west rtt and come up with bug report, of it not work. 
I will close this PR though.